src/NuGet.Core/NuGet.Versioning/VersionRangeComparer.cs

```diff
@@ -82,7 +98,7 @@ public bool Equals(VersionRangeBase? x, VersionRangeBase? y)
         /// Creates a hash code based on all properties of the range. This follows the rules of the
         /// version comparer when comparing the version bounds.
         /// </summary>
-        public int GetHashCode(VersionRangeBase obj)
+        public int GetHashCode(VersionRangeBase? obj)
```

Comment:
This doesn't look correct to me. If you were to (temporarily, as a test) add `net7.0` to NuGet.Versioning's TagetFramework, I would expect the nullability analyzer to warn you that the nullability of this method does not match the inherited signature (object.GetHashCode).

Reply:
I'm on a phone so can't easily check right now. You can widen but not narrow from the base implementation, so this should be fine.